server/webhook_jira.go

```diff
@@ -29,6 +31,50 @@ type JiraWebhook struct {
 	IssueEventTypeName string `json:"issue_event_type_name"`
 }
 
+func (jwh *JiraWebhook) expandIssue(p *Plugin, instanceID types.ID) error {
+	instance, err := p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		return err
+	}
+
+	// Jira Cloud comment event. We need to fetch issue data because it is not expanded in webhook payload.
+	isCommentEvent := jwh.WebhookEvent == commentCreated || jwh.WebhookEvent == commentUpdated || jwh.WebhookEvent == commentDeleted
+	if isCommentEvent && instance.Common().IsCloudInstance() {
+		if _, ok := instance.(*cloudInstance); ok {
+			issue, err := p.getIssueDataForCloudWebhook(instance, jwh.Issue.ID)
+			if err != nil {
+				return err
+			}
+
+			jwh.Issue = *issue
+		} else if _, ok := instance.(*cloudOAuthInstance); ok {
+			mmUserID, err := p.userStore.LoadMattermostUserID(instanceID, jwh.Comment.Author.AccountID)
+			if err != nil {
+				return err
```

Comment:
I think we should return a more specific error here. Something like:

```suggestion
				return errors.Wrap(err, "Jira comment author is not connected to Mattermost. Cannot create subscription posts for this comment.")
```